jaraco/abode/event_controller.py

```diff
@@ -169,7 +170,7 @@ def socketio(self):
 
     def _on_socket_started(self):
         """Socket IO startup callback."""
-        cookies = self._client._get_session().cookies.get_dict()
+        cookies = copy.copy(self._client._get_session().cookies.shelf.store)
```

Comment:
Thanks for demoing this fix. I don't think we want to rely on internal implementation details of the cookies. Maybe the cookie jar should expose a `get_dict()`? Also, I'd like to add some tests. Even if we can't test the socket interface, I'd like to test the interfaces on which the socket interface relies (i.e. generate cookie string from session cookies).

Reply:
Why was `copy` necessary?

Reply:
I looked briefly into adding `get_dict()` support to `jaraco.net.cookies.ShelvedCookieJar`, but stumbled when tests were failing to add two different cookies, so there may be some more underlying problems with the cookie implementation. I'll need to investigate. In the meantime, let's go with this implementation after adding some tests to capture the expectation (I'll do that).

Reply:
Generally I agree about not relying on internal implementation details, but `shelf` and `store` are both public attributes. If they're supposed to be internal, I'd suggest prefixing them and adding a `get_dict()` method to the cookie jar.

I used `copy()` just in case another thread might be modifying the dict as we're trying to read it. Might be unnecessarily cautious but I'd rather not cause weird problems later.

Reply:
In e39806e, I've extracted a method that should render cookies in the jar. I'm not sure it's correct, because I had to render `cookie.value` to get something that didn't look like `foo=<Cookie foo=bar for any.local/>`. Please take a look.

Reply:
LGTM as long as it passes the tests.